synapse/rest/admin/users.py

```diff
@@ -1097,3 +1097,28 @@ async def on_DELETE(
         await self.store.delete_ratelimit_for_user(user_id)
 
         return 200, {}
+
+
+class UsernameAvailableRestServlet(RestServlet):
```

Comment:
we ought to break up this file into smaller modules. Also, there ought to be a "register servlets" function in here so that they don't all have to be imported individually into `synapse.rest.admin.__init__`.

Still, I guess this straw isn't going to break the camel's back.

Reply:
I've broken out both the tests and the rest handler, as both were easy to move :)